resources/services/memorystore/gen.hcl

```diff
@@ -0,0 +1,214 @@
+  }
+
+  column "host" {
+    description = "Hostname or IP address of the exposed Redis endpoint used by clients to connect to the service"
+  }
+
+  column "labels" {
+    description = "Resource labels to represent user provided metadata"
+  }
+
+  column "location_id" {
+    description = "The zone where the instance will be provisioned"
+  }
+
+  column "maintenance_policy" {
+    description       = "The maintenance policy for the instance"
+    type              = "json"
+    generate_resolver = true
+  }
+
+  column "maintenance_schedule" {
+    description       = "Date and time of upcoming maintenance events which have been scheduled"
+    type              = "json"
+    generate_resolver = true
+  }
+
+  column "memory_size_gb" {
+    description = "Redis memory size in GiB"
+  }
+
+  column "name" {
+    description = "Unique name of the resource"
+  }
+
+  column "persistence_config" {
+    description       = "Persistence configuration parameters"
+    type              = "json"
+    generate_resolver = true
+  }
+
+  column "persistence_iam_identity" {
+    description = "Cloud IAM identity used by import / export operations to transfer data to/from Cloud Storage"
+  }
+
+  column "port" {
+    description = "The port number of the exposed Redis endpoints"
+  }
+
+  column "read_endpoint" {
+    description = "Hostname or IP address of the exposed readonly Redis endpoint"
+  }
+
+  column "read_endpoint_port" {
+    description = "The port number of the exposed readonly redis endpoint"
+  }
+
+  column "read_replicas_mode" {
+    description = "Read replicas mode for the instance"
+  }
+
+  column "redis_configs" {
+    description = "Redis configuration parameters"
+  }
+
+  column "redis_version" {
+    description = "The version of Redis software"
+  }
+
+  column "replica_count" {
+    description = "The number of replica nodes"
+  }
+
+  column "reserved_ip_range" {
+    description = "IP range for node placement"
+  }
+
+  column "secondary_ip_range" {
+    description = "Additional IP range for node placement"
+  }
+
+  column "state" {
+    description = "The current state of the instance"
+  }
+
+  column "status_message" {
+    description = "Additional information about the current status of the instance, if available"
+  }
+
+  column "suspension_reasons" {
+    description = "Reasons that causes instance in `SUSPENDED` state"
+  }
+
+  column "tier" {
+    description = "The service tier of the instance"
+  }
+
+  column "transit_encryption_mode" {
+    description = "The TLS mode of the Redis instance"
+  }
+
+  column "nodes" {
+    description       = "Redis instance nodes properties"
+    type              = "json"
+    generate_resolver = true
+  }
+
+  relation "gcp" "memorystore_redis_instance" "server_ca_certs" {
+    description = "List of server CA certificates for the instance"
+
+    options {
+      primary_keys = [
+        "redis_instance_cq_id",
```

Comment:
I don't know the answer, but maybe @amanenk can tell us - is it okay to use a `cq_id` as part of a primary key? Does it matter, or are there better options here?

Reply:
left only `redis_instance_cq_id `

Reply:
I was actually wondering whether it's okay to use `redis_instance_cq_id` as the primary key, or not 🤔 I haven't seen `*cq_id` being used as primary keys very much, if at all.